### ISSUE TITLE
nushell doesn't execute binary although it is in path (windows)

### ISSUE THREAD

**Issue by NiklasJonsson** [nushell/nushell#3534]
**Describe the bug**
When I upgraded to nushell 0.31 I can no longer run my cmake executable even though it's directory is in path. This worked in nushell 0.28. I also tried 0.30 where it fails as well. Adding the .exe suffix does not make a difference either. The `which` command does find it:

` which cmake`
```
───┬───────┬──────────────────────────────────────┬─────────
 # │  arg  │                 path                 │ builtin
───┼───────┼──────────────────────────────────────┼─────────
 0 │ cmake │ C:\Program Files\CMake\bin\cmake.EXE │ false
───┴───────┴──────────────────────────────────────┴─────────
```

**To Reproduce**
1. Install cmake (I did it via chocolatey)
2. Run 'cmake' in the prompt
3. The following is printed
`'cmake' is not recognized as an internal or external command,
operable program or batch file.`

**Expected behavior**
nushell finds the executable and runs it.

**Configuration**

```
> version | pivot
version | pivot
───┬────────────────────┬────────────────────────────────────────────────────────────────
 # │      Column0       │                            Column1
───┼────────────────────┼────────────────────────────────────────────────────────────────
 0 │ version            │ 0.31.0
 1 │ build_os           │ windows-x86_64
 2 │ rust_version       │ rustc 1.54.0-nightly (5c0292654 2021-05-11)
 3 │ rust_channel       │ nightly-x86_64-pc-windows-msvc (default)
 4 │ cargo_version      │ cargo 1.54.0-nightly (e51522ab3 2021-05-07)
 5 │ pkg_version        │ 0.31.0
 6 │ build_time         │ 2021-06-01 13:14:24
 7 │ build_rust_channel │ release
 8 │ features           │ ctrlc, default, directories, dirs, rustyline, term, which, zip
───┴────────────────────┴────────────────────────────────────────────────────────────────
```
`echo $nu.path` contains the following line:
 34 │ C:\Program Files\CMake\bin
which contain the binary I want to run.

I realize this bug report is a bit broad, I will experiment some more to try to narrow down the issue, like trying with 0.29 as well.

Edit: 0.29 works as well.

**Comment by NiklasJonsson** [comment]
I can confirm that 0.29 works as well.

**Comment by fdncred** [comment]
@NiklasJonsson Can you try in 0.32. I had the same issue in 0.31 and it is fixed for me in 0.32.

**Comment by NiklasJonsson** [comment]
Yes, this is fixed in 0.32 for me as well. Thank you! Closing.
